src/syntax_analysis.rs

```diff
@@ -0,0 +1,192 @@
+use std::collections::HashMap;
+use syntect::parsing::syntax_definition::{
+    ContextReference, MatchOperation, MatchPattern, Pattern, SyntaxDefinition,
+};
+use syntect::parsing::{Scope, SyntaxSet, SyntaxSetBuilder};
+
+/// Used to look up (by name) what dependencies a given [SyntaxDefinition] has
+type SyntaxToDependencies = HashMap<String, Vec<Dependency>>;
+
+/// Used to look up what [SyntaxDefinition] that corresponds to a given [Dependency]
+type DependencyToSyntax<'a> = HashMap<Dependency, &'a SyntaxDefinition>;
+
+/// Represents a dependency on an external `.sublime-syntax` file.
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+enum Dependency {
+    /// By name. Example YAML: `include: C.sublime-syntax`
+    ByName(String),
+
+    /// By scope. Example YAML: `embed: scope:source.c`
+    ByScope(Scope),
+}
+
+/// Change to true to make syntax dependency analysis print more details of what it is seeing
+const VERBOSE: bool = false;
```

Comment:
I think I would prefer a runtime switch here (a `verbose: bool` member in `SyntaxSetDependencyBuilder`).

Or to get rid of it completely by either removing the verbose=true path or by always being verbose. But only if the verbose-mode output is still readable (haven't tested it yet)

Reply:
It turns out that the "unlinked context" code from Keith prints mostly the same thing as my verbose output, so I'll just remove it from my code. And from what I understand, when we differ, it is because it is a non-fatal dependency that is missing.

For the record, my code warned about these extra missing dependencies that Keith code did not:

```
WARNING: No syntax found ByName("OpenMP")
WARNING: No syntax found ByScope(<source.js.css>)
WARNING: No syntax found ByScope(<source.lean.markdown>)
WARNING: No syntax found ByScope(<text.dart-doccomments>)
WARNING: No syntax found ByScope(<text.html.php>)
```

I'll unconditionally keep
```
eprintln!("WARNING: Unknown dependencies for {}", name);
```
though, because that shall never happen.